Diagnostics/Get-VSwitchPathCost.ps1

```diff
@@ -127,34 +136,25 @@ function Get-CounterAverage([String[]] $statArray)
 
 if (IsHyperThreadingEnabled)
 {
-    Log "ERROR: Hyper-Threading is enabled. Please disable Hyper-Threading before starting the test."
+    Log "WARNING: Hyper-Threading is enabled. This may affect the accuracy of the results."
 }
 
-$switch = Get-VMSwitch $SwitchName
-if ($switch -eq $null)
+if ($BaseCpuNumber -gt $(Get-MaxCpuNumber))
 {
-    Log "ERROR: Switch $SwitchName does not exist"
-}
-
-if ($switch.Count -gt 1)
-{
-    Log "ERROR: There are more than one vswitch with the name $SwitchName"
+    Log "ERROR: BaseCpuNumber is greater than Max CPU Number."
 }
 
 if (-not $PSBoundParameters.ContainsKey("MaxCpuNumber"))
 {
-    $cpuPerNuma = GetNumCpusPerNUMA
-    $numaCount = GetNumaCount
-    $MaxCpuNumber = $cpuPerNuma * $numaCount - 1
+    $MaxCpuNumber = Get-MaxCpuNumber
     Log "Set MaxCpuNumber to $MaxCpuNumber"
 }
-
-if ($MaxCpuNumber -lt $BaseCpuNumber)
+elseif (IsHyperThreadingEnabled)
 {
-   Log "ERROR: Max CPU number is less than base CPU number"
+    # force MaxCpuNumber to odd value with HT so the entire core is measured
```

Comment:
This is correct, right?

Reply:
No, with HyperThreading enabled, we can't really deterministically measure the CPU cycles. 
Also, the maxCpuNumber calculation below is not correct.  This means you're ignoring half of the CPUs.  The HT core pairs are 0/1, 2/3, 4/5, etc.